packages/react/src/input.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from "@hpcc-js/preact-shim";
+
+export interface IInput {
+    type: "number" | "range" | "date" | "text" | "select" | "checkbox";
+    value: number | string | boolean;
+    className?: string;
+    attributes?: any;
+    callbacks?: any;
+}
+
+export const Input: React.FunctionComponent<IInput> = ({
+    type = "text",
+    className,
+    value,
+
+    attributes = {}
+}) => {
+    return <input
+        value={value}
+        {...attributes}
+        type={type}
+        className={className}
+    />;
+};
+
+export interface ILabelInput {
+    label: string;
+    input: IInput;
+    labelWidth?: number;
+    labelMinWidth?: number;
+    labelMaxWidth?: number;
+    width?: number;
+    minWidth?: number;
+    maxWidth?: number;
+    flexDirection?: "row" | "column";
+}
+
+export const LabelInput: React.FunctionComponent<ILabelInput> = ({
+    label = "",
+    width,
+    minWidth = 60,
+    maxWidth = 240,
+    labelWidth,
+    labelMinWidth = 30,
+    labelMaxWidth = 120,
+    flexDirection = "row",
+    input
+}) => {
+    const i18n = label => label;
+    const inputs = [];
+    if (input.type === "number" || input.type === "range") {
+        inputs.push(<Input {...input}/>);
+        inputs.push(<Input {...input} type="range"/>);
+    } else {
+        if (input.type === "checkbox" && input.attributes && typeof input.attributes.checked === "undefined") {
+            input.attributes.checked = input.value;
+        }
+        inputs.push(<Input {...input}/>);
+    }
+
+    let wrapperStyle = `display:flex;flex-direction:${flexDirection};`;
+    if (width)wrapperStyle += `width:${width}px;`;
+    if (minWidth)wrapperStyle += `min-width:${minWidth}px;`;
+    if (maxWidth)wrapperStyle += `max-width:${maxWidth}px;`;
+
+    let spanStyle = "overflow:hidden;text-overflow:ellipsis;";
+    if (labelWidth)spanStyle += `width:${labelWidth}px;`;
+    if (labelMinWidth)spanStyle += `min-width:${labelMinWidth}px;`;
+    if (labelMaxWidth)spanStyle += `max-width:${labelMaxWidth}px;`;
+
+    return <label style={wrapperStyle} alt={i18n(label)}>
+        <span style={spanStyle}>{label}</span>
+        {inputs}
+    </label>;
+};
+
+export interface ILabelInputGroup {
+    inputs: ILabelInput[];
+}
+
+export const LabelInputGroup: React.FunctionComponent<ILabelInputGroup> = ({
+    inputs = []
+}) => {
+    const items = [];
+    inputs.forEach(input => {
+        items.push(
+            <LabelInput
+                {...input}
+            />
+        );
+    });
+    const divStyle = "display:flex;flex-direction:column;";
+    return <div style={divStyle}>
+        {items}
```

Comment:
When there are lists of items like this, they should have "keys" to improve performance.